src/charm.py

```diff
@@ -1006,9 +1091,140 @@ def restart_backup_service(self) -> None:
         charmed_mongodb_snap = snap_cache["charmed-mongodb"]
         charmed_mongodb_snap.restart(services=["pbm-agent"])
 
+    def _scope_obj(self, scope: Scopes):
+        if scope == APP_SCOPE:
+            return self.app
+        if scope == UNIT_SCOPE:
+            return self.unit
+
+    def _peer_data(self, scope: Scopes):
+        if not self._peers:
+            return {}.setdefault(scope, {})
+        scope_obj = self._scope_obj(scope)
+        return self._peers.data[scope_obj]
+
+    @staticmethod
+    def _compare_secret_ids(secret_id1: str, secret_id2: str) -> bool:
+        """Reliable comparison on secret equality.
+
+        NOTE: Secret IDs may be of any of these forms:
+         - secret://9663a790-7828-4186-8b21-2624c58b6cfe/citb87nubg2s766pab40
+         - secret:citb87nubg2s766pab40
+        """
+        if not secret_id1 or not secret_id2:
+            return False
+
+        regex = re.compile(".*[^/][/:]")
+
+        pure_id1 = regex.sub("", secret_id1)
+        pure_id2 = regex.sub("", secret_id2)
+
+        if pure_id1 and pure_id2:
+            return pure_id1 == pure_id2
+        return False
```

Comment:
Instead of this function now we should be able to use `secret1.unique_id == secret2.unique_id`

Reply:
I prefer for now to keep the code same in k8s and vm charms and do this change later

Reply:
In which version of ops this is available?